File: SharpLinker/Services/ComunicationResolver.cs
using SharpLinker.Enums;
using SharpLinker.Models;
using SharpLinker.Services.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SharpLinker.Services
{
    public class ComunicationResolver : IComunicationResolver
    {
        Dictionary<PageActions, Func<Message, Task> > eventHandlers;

        public ComunicationResolver()
        {
            eventHandlers = new Dictionary<PageActions, Func<Message, Task>>();
        }

        public async Task Resolve(Message data)
        {
            await eventHandlers[data.Action](data);
        }

        public void Subscribe(PageActions action, Func<Message, Task> func)
        {
            eventHandlers[action] = func;
        }

        public void Unsubscribe(PageActions action)
        {
            eventHandlers[action] = null; 
        }
    }
}

File: SharpLinker-Extension/assets/ts/services/comunicationResolver.ts

class ComunicationResolver{
    eventFunctions : ActionsMap = {};

    resolve(event : any){
        if(this.eventFunctions[event.action] == undefined){
            console.error('There is no method to resolve action: ' + event.action);
            return;
        }
        this.eventFunctions[event.action](event.data);
    }

    subscribe(action, func){
        this.eventFunctions[action] = func;
    }

    unsubscribe(action, func){
        this.eventFunctions[action] = null;
    }
} 

interface ActionsMap {
    [name: string]: Function;
}